### ISSUE TITLE
ParameterSet support 

### ISSUE THREAD

**Issue by davidlange6** [cms-sw/cmssw#20161]
asked in #20144 

@Dr15Jones @wddgit can you think of any way to support
```c++
template <typename T, std::size_t N>
std::array<T, N>
ParameterSet::getParameter<std::array<T, N>>(...)
```
directly in `ParameterSet` ?

I I understand that it cannot be done as an overload or partial specialisation of `ParameterSet::getParameter<T>(...)` .

**Comment by davidlange6** [comment]
assign core

**Comment by Dr15Jones** [comment]
How about just a helper function that takes the array's template arguments and the ParameterSet? Internally it could get an std::vector of the correct type and throw an exception if the python list is the wrong size.

**Comment by Dr15Jones** [comment]
Solved via #22362

**Comment by Dr15Jones** [comment]
+1
